Jasonbreen tim chang87/mid 97 housekeeping after merge

jason-breen opened the pull request:
`package.json`
 - Updated start script to incorporate electron-forge changes. Removed start:forge.

`src/App.tsx`
 - Refactored to pass string 'directory' or 'file' to openFile handler. (see `src/Types.ts`)
 - Added `copyConfig()` function and associated button to enable user selection of config file

`src/Types.ts`
 - Updated `openFile:` to reflect new string parameter
   - This allows the handler to be used for selection of either a directory or individual file
 - Added copyConfig method for configuration management

`src/configManager.ts`
 - Created new module to allow user specified config file to be saved and accessed by the application
 - `determineDir` uses environment variables to determine the appropriate application data folder
 - `copyConfig` copies the user specified config file to the app data folder
 - `readConfig` reads the config stored in the application data folder and returns the configuration object

`src/dbModels.ts`
 - Refactored to utilize the configuration manager module for MongoDB_URI

`src/index.ts` & `src/preload.ts`
 - Commented out the proxy on port 9000 to prevent conflict with middle-aware agent
 - Added handler for configManager functions

`src/parseAPI-FE-Req.ts`
 - Cleanup - added '?' to object access for scenarios where options are not provided

`src/proxyServer.ts`
 - Imported `readConfig` to utilize user configuration to set the proxy port for middle-aware

jason-breen committed:
Housekeeping updates made:
 - package.json: moved start:forge script to start
 - parseAPI-FE-Eq.ts: added ? to options.method for cases where the options object does not exist
 - dbModels.ts: Removed MongoDB API Key from file. Troubleshooting dotenv led to the following refactor
 - WIP configManager: Created configuration manager for user to select project config file, which will contain db credentials.
   Note that the above is a work in progress. Currently have button working on front end, but functionality has not been verified.

jason-breen committed:
Complete configManager feature
 - Working for MongoDB_URI for database functionality

jason-breen committed:
Minor refactor for configManager
 - Exported each function and a default that contains all functions to allow for import of individual configManager methods when necessary

timchang87 reviewed (commented):
Getting this error

`[0] App threw an error during load
[0] Error: ENOENT: no such file or directory, open '/Users/timchang/Library/Application Support/middle-aware/config.json'
[0]     at Object.openSync (node:fs:600:3)
[0]     at Object.func [as openSync] (node:electron/js2c/asar_bundle:2:1822)
[0]     at Object.readFileSync (node:fs:468:35)
[0]     at Object.t.readFileSync (node:electron/js2c/asar_bundle:2:9165)
[0]     at Object.readConfig (/Users/timchang/Documents/Codesmith/OSP/middle-aware/.webpack/main/webpack:/middle-aware/src/configManager.ts:33:1)
[0]     at Object../src/dbModels.ts (/Users/timchang/Documents/Codesmith/OSP/middle-aware/.webpack/main/webpack:/middle-aware/src/dbModels.ts:33:1)
[0]     at __webpack_require__ (/Users/timchang/Documents/Codesmith/OSP/middle-aware/.webpack/main/webpack:/middle-aware/webpack/bootstrap:19:1)
[0]     at Object../src/dbController.ts (/Users/timchang/Documents/Codesmith/OSP/middle-aware/.webpack/main/webpack:/middle-aware/src/dbController.ts:13:20)
[0]     at __webpack_require__ (/Users/timchang/Documents/Codesmith/OSP/middle-aware/.webpack/main/webpack:/middle-aware/webpack/bootstrap:19:1)
[0]     at Object../src/index.ts (/Users/timchang/Documents/Codesmith/OSP/middle-aware/.webpack/main/webpack:/middle-aware/src/index.ts:21:12)
^C[1] npx tailwindcss -i ./src/App.css -o ./src/output.css --watch exited with code SIGINT
[0] electron-forge start exited with code 127`

jason-breen committed:
Corrected error on launch
App was not loading if config file did not exist. The following changes are made to correct the condition:
configManager.ts
 - Added conditional statement to create default config upon call to readConfig method if one is not available.

dbModels.ts
 - Created mongoConnect function and exported to create connection after config file is uploaded.
 - readConfig is called within this function and mongoose.connect is only called if a MongoDB_URI has been provided in the config file.

index.ts
 - Import mongoConnect. Invoke mongoConnect after config file is uploaded.
